frontend/lib/src/hooks/useOnInputChange.test.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { waitFor } from "@testing-library/react"
+import { renderHook } from "@testing-library/react-hooks"
+
+import useOnInPutChange from "./useOnInputChange"
+
+describe("useOnInputChange", () => {
```

Comment:
**suggestion (non-blocking):** I'm glad we have unit tests for hooks! We could make these even more effective if we were to write a resuable set of tests that we can apply to the elements themselves so that it more accurately reflects how the users use the software. This suggestion would be more inline with the [guiding principles of React Testing Library](https://testing-library.com/docs/guiding-principles)

> [The more your tests resemble the way your software is used, the more confidence they can give you.](https://twitter.com/kentcdodds/status/977018512689455106)

With that said, this is not blocking feedback for this PR, as we are also covering this behavior in e2e tests. Just something to think about in general when it comes to approach writing frontend tests in order to better target user behavior over implementation details.

Reply:
Yeah I love the idea of reusable set of tests! We should definitely tackle that, but I agree that it is probably a project on its own to create the test suite and update our unit / e2e tests to leverage those